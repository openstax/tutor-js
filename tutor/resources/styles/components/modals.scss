.modal-dialog,
.modal {
  $header-padding: 30px;
  .modal-header {
    padding: $header-padding/2 $modal-padding;
    font-size: 16px;
    background-color: $tutor-neutral-bright;
    border-bottom: 1px solid $tutor-neutral-light;
    align-items: center;
    justify-content: space-between;
  }
  .modal-content {
    border-radius: 0;
    border: 0;
  }

  .modal-body {
    padding: $modal-padding;
    background-color: white;
    line-height: 1.6;
    p.lead {
      font-size: 1.75rem;
      line-height: 2.8rem;
      font-weight: 300;
    }
  }

  .modal-footer {
    padding: 0 $modal-padding $modal-padding $modal-padding;
    border-top: 0;
    display: flex;
  }

}

@import './modals/warning';
@import './modals/error';
@import './modals/terms';
@import './modals/onboarding';
