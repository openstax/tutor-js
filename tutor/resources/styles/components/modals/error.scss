.error.modal .modal-header,
.modal-header.error {
  background-color: #f5e9ea;
  color: #b33238;
  align-items: center;
  justify-content: flex-start;
  .ox-icon {
    font-size: 24px;
    fill: $tutor-danger;
    margin-right: 0.5rem;
  }
}

.error.modal {
  &.show {
    .modal-dialog {
      transform: translateY(110px);
    }
  }
}
