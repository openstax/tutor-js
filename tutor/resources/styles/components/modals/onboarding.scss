.onboarding {
  @include ost-hero-modal();
}

.onboarding.modal {
  @media screen and ( max-width: $mobile-collapse-breakpoint ){
    height: 100%;
  }
  .modal-dialog {
    width: 500px;
    margin: 0 auto;
    @media screen and ( max-width: $mobile-collapse-breakpoint ){
      width: 100%;
    }
  }
  &.show .modal-dialog {
    transform: translateY(130px);
    @media screen and ( max-width: $mobile-collapse-breakpoint ){
      transform: translateY(0) !important;
      height: 100%;
    }
  }
  .modal-content {
    border-radius: 0;
    @media screen and ( max-width: $mobile-collapse-breakpoint ){
      background-color: #f5f5f5;
      height: 100%;
    }
  }
  &.pay-now-or-later,
  &.free-trial-ended,
  &.make-payment {
    .modal-dialog {
      transform: translateY(80px);
    }
  }

  .onboarding-nag {
    background-color: $tutor-neutral-lighter;

    .heading {
      font-weight: 700;
      font-size: 18px;
      padding: 50px 50px 0 50px;
    }

    .body {
      font-weight: 500;
      font-size: 16px;
      line-height: 24px;
      padding: 50px;
      @media screen and ( max-width: $mobile-collapse-breakpoint ){
        padding: 50px 0;
      }
    }

    .footer {
      padding: 0 50px 50px 50px;
      .btn {
        width: 100%;
        margin-top: 0.5rem;
        & + .btn { margin-left: 0; }
      }
      &.got-it {
        .btn { width: inherit; }
      }
    }

    &.freshly-created-course {
      .btn-link { padding: 1rem 0; }
    }

    &.free-trial-ended,
    &.free-trial-activated
    {
      .footer {
        .btn { width: inherit; }
      }
    }

    &.pay-now-or-later,
    &.payments-disabled {
      padding: 50px;
      background-color: $tutor-neutral-bright;
      text-align: center;
      .btn.later {
        background-color: $tutor-neutral;
        color: $tutor-neutral-lightest;
      }
      p {
        font-size: 1.4rem;
        margin: 20px 0;

      }
      .or {
        overflow: hidden;
        text-align: center;
        margin: 20px 0;
        &:before,
        &:after {
          background-color: $tutor-neutral-light;
          content: "";
          display: inline-block;
          height: 1px;
          position: relative;
          vertical-align: middle;
          width: 50%;
        }
        &:before {
          right: 0.5em;
          margin-left: -50%;
        }
        &:after {
          left: 0.5em;
          margin-right: -50%;
        }
      }
    }
    &.payments-disabled {
      h3 { margin-bottom: 20px; }
    }
  }
}
