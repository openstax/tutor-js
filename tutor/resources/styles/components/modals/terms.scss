.user-terms {
  @include ost-hero-modal();
  @media (max-width: $mobile-collapse-breakpoint) {
    // overriding the inline style
    padding-left: 0 !important;
    padding-right: 0 !important;
  };
}

.user-terms.modal {
  $spacing: 20px;

  .modal-dialog {
    opacity: 1;
    transition: opacity .4s ease-in-out;
    @media (max-width: $mobile-collapse-breakpoint) {
      height: 100%;
      margin: 0;
      max-width: inherit;
    };
  }
  .modal-content {
    background-color: $tutor-neutral-bright;
    @media (max-width: $mobile-collapse-breakpoint) {
      height: 100%;
    };
  }
  .modal-header {
    font-size: 125%;
    font-weight: bold;
    border-bottom: 0;
    padding: $spacing $spacing*2;

    @media (max-width: $mobile-collapse-breakpoint) {
      display: flex;
      flex-flow: column;

      .course-branding {
        font-size: 12px;
      }
      .header-title {
        margin-top: 15px;
        font-size: 22px;
      }
    };
  }
  .modal-body {
    max-height: 50vh;
    overflow-y: auto;
    overflow-x: hidden;
    padding: $spacing*0.5 $spacing*1.5;
    margin: 0 $spacing*2;
    border: 1px solid $tutor-neutral-light;
    background-color: $tutor-white;
    h1 { font-size: 125%; }
    h2 { font-size: 120%; }
    h3 { font-size: 115%; }

    @media (max-width: $mobile-collapse-breakpoint) {
      margin: 0 $mobile-margin;
      padding: 10px $mobile-margin;
    };
  }

  .modal-footer {
    border-top: 0;
    text-align: left;
    padding: $spacing $spacing*2 $spacing*2 $spacing*2;

    @media (max-width: $mobile-collapse-breakpoint) {
      padding: 2rem;
      justify-content: center;
    };

    .btn {
      font-size: 20px;
      font-weight: normal;
      padding: 1rem 2rem;

      @media (max-width: $mobile-collapse-breakpoint) {
        width: 75%;
      };
    }
  }
  &.show {
    .modal-dialog {
      transform: translateY(110px);
      @media (max-width: $mobile-collapse-breakpoint) {
        transform: translateY(0);
      };
    }
  }
  &.is-loading {
    .modal-dialog { opacity: 0; }
    .modal-footer {
      .btn { visibility: hidden; }
    }
  }
}
