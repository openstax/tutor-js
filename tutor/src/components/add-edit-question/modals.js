import { React, PropTypes, styled, observer, useState, useEffect } from 'vendor';
import { Button, Modal } from 'react-bootstrap';
import { map, find, isEmpty } from 'lodash';
import { colors } from 'theme';
import { Question, ArbitraryHtmlAndMath } from 'shared';
import AddEditQuestionUX from './ux';
import QuestionModel from '../../../../shared/src/model/exercise/question';

const PreviewModal = styled(Modal)`
  .modal-dialog {
    min-width: 800px;
  }
  .modal-body {
    .context {
      color: ${colors.neutral.thin};
    }
    textarea {
      width: 100%;
      min-height: 10.5em;
      line-height: 1.5em;
      margin: 2.5rem 0 0 0;
      padding: 0.5em;
      border: 1px solid ${colors.neutral.std};
      color: ${colors.neutral.dark};
      background-color: ${colors.neutral.cool};
    }
    .btn.btn-primary {
      float: right;
      margin-top: 1rem;
    }
  }
`;

const StyledHeader = styled(Modal.Header)`
  font-weight: bold;
  font-size: 1.8rem;
  padding: 2.5rem 3rem;
`;

const StyledBody = styled(Modal.Body)`
  font-size: 1.6rem;
  line-height: 2.5rem;
  .note {
      font-size: 1.4rem;
      color: ${colors.neutral.thin};
  }
`;

const ControlsWrapper = styled.div`
  margin-top: 3rem;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`;

const Controls = styled.div`
  .btn.btn-default {
    background-color: white;
    color: ${colors.neutral.grayblue};
  }
  .btn + .btn {
    margin-left: 1.5rem;
  }
`;


const FeedbackTipModal = observer(({ ux }) => {
    return (
        <Modal
            show={ux.feedbackTipModal.show}
            backdrop="static"
        >
            <StyledHeader>
        Tip: Add feedback!
            </StyledHeader>
            <StyledBody>
                <p>Add explanatory feedback for the answer choices. Students learn better when they recieve immediate and specific feedback.</p>
                <p className="note">Note: Ideally, this feedback should anticipate common misconceptions or calculation errors.</p>
                <ControlsWrapper>
                    <Controls>
                        <Button variant="default" size="lg" onClick={() => ux.doPublish(ux.feedbackTipModal.shouldExitOnPublish)}>
              Publish anyway
                        </Button>
                        <Button variant="primary" size="lg" onClick={() => ux.feedbackTipModal.show = false}>
              Add feedback
                        </Button>
                    </Controls>
                </ControlsWrapper>
            </StyledBody>
        </Modal>
    );
});
FeedbackTipModal.propTypes = {
    ux: PropTypes.instanceOf(AddEditQuestionUX).isRequired,
};

const ExitWarningModal = observer(({ ux }) => {
    return (
        <Modal
            show={ux.showExitWarningModal}
            backdrop="static"
        >
            <StyledHeader>
        Exit without publishing?
            </StyledHeader>
            <StyledBody>
                <p>The question is not published yet and will not be saved. Are you sure you want to exit this form?</p>
                <ControlsWrapper>
                    <Controls>
                        <Button variant="default" size="lg" onClick={() => {
                            ux.showExitWarningModal = false;
                            ux.onDisplayModal(false);
                        }}>
                Yes, exit
                        </Button>
                        <Button variant="primary" size="lg" onClick={() => ux.showExitWarningModal = false}>
                Cancel
                        </Button>
                    </Controls>
                </ControlsWrapper>
            </StyledBody>
        </Modal>
    );
});
ExitWarningModal.propTypes = {
    ux: PropTypes.instanceOf(AddEditQuestionUX).isRequired,
};

const CoursePreviewOnlyModal = observer(({ onDisplayModal }) => {
    return (
        <Modal
            show={true}
            backdrop="static"
            onHide={() => onDisplayModal(false)}>
            <StyledHeader closeButton>
        This is a demo course!
            </StyledHeader>
            <StyledBody>
                <p>Adding or editing a question is <strong>not allowed</strong> in a demo course. 
            You can create a live course to add or edit questions.</p>
                <ControlsWrapper>
                    <Controls>
                        <Button variant="primary" size="lg" onClick={() => onDisplayModal(false)}>
                Okay
                        </Button>
                    </Controls>
                </ControlsWrapper>
            </StyledBody>
        </Modal>
    );
});
CoursePreviewOnlyModal.propTypes = {
    onDisplayModal: PropTypes.func.isRequired,
};

const QuestionPreviewModal = observer(({ ux }) => {
    // dont show answers yet when it is a two-step question
    const [showAnswers, setshowAnswers] = useState(!ux.isTwoStep);
    const [freeResponseText, setFreeReponseText] = useState('');

    useEffect(() => {
        setshowAnswers(!ux.isTwoStep);
    }, [ux.isTwoStep]);

    const question = new QuestionModel({
        stem_html: ux.questionText,
        answers: map(ux.filledOptions, o => {
            return {
                id: o.id,
                content_html: o.text,
                correctness: o.isCorrect ? '1.0' : '0.0',
                feedback_html: o.feedback,
            };
        }),
        formats: [ux.isMCQ ? 'multiple-choice' : 'free-response'],
    });
    //only OS questions has context
    const context = () => {
        if (!ux.fromExercise || !ux.fromExercise.context) return null;
        return (
            <ArbitraryHtmlAndMath className="context" block={true} html={ux.fromExercise.context} />
        );
    };

    let QuestionComponent;
    if (ux.isMCQ) {
        const correctAnswer = find(ux.filledOptions, o => o.isCorrect);
        QuestionComponent =
      <>
          <Question
              question={question}
              answer_id={correctAnswer ? correctAnswer.id : null}
              correct_answer_id={correctAnswer ? correctAnswer.id : null}
              hideAnswers={!showAnswers}
          />
          {
              !showAnswers &&
          <>
              <textarea
                  placeholder="Enter your response..."
                  aria-label="question response text box"
                  value={freeResponseText}
                  onChange={({ target: { value } }) => setFreeReponseText(value)} />
              <Button
                  size="lg"
                  disabled={isEmpty(freeResponseText.trim())}
                  onClick={() => setshowAnswers(true)}
              >
              Submit
              </Button>
          </>
          }
        
      </>;
    }
    else {
        QuestionComponent = 
      <>
          <Question question={question}/>
          <textarea
              placeholder="Enter your response..."
              aria-label="question response text box"
              readOnly />
      </>
        ;
    }

    return (
        <PreviewModal
            show={ux.showPreviewQuestionModal}
            backdrop="static"
            onHide={() => ux.showPreviewQuestionModal = false}>
            <StyledHeader closeButton>
        Question Preview
            </StyledHeader>
            <StyledBody>
                {context()}
                {QuestionComponent}
            </StyledBody>
        </PreviewModal>
    );
});

const DeleteExerciseModal = ({ show, onHide, onDelete }) => {
    return (
        <Modal
            show={show}
            backdrop="static"
        >
            <StyledHeader>
          Delete Question?
            </StyledHeader>
            <StyledBody>
                <p>Are you sure you want to permanently delete this question? Question <strong>will not</strong> be deleted from any open assignments.</p>
                <p>You canâ€™t undo this action.</p>
                <p className="note">Tip: To exclude a bad question from an open assignment, drop it from the assignment details page.</p>
                <ControlsWrapper>
                    <Controls>
                        <Button variant="default" size="lg" onClick={onDelete}>
                  Yes, delete
                        </Button>
                        <Button variant="primary" size="lg" onClick={onHide}>
                  Cancel
                        </Button>
                    </Controls>
                </ControlsWrapper>
            </StyledBody>
        </Modal>
    );
};
DeleteExerciseModal.propTypes = {
    show: PropTypes.bool.isRequired,
    onHide: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired,
};

export {
    FeedbackTipModal,
    ExitWarningModal,
    CoursePreviewOnlyModal,
    QuestionPreviewModal,
    DeleteExerciseModal };
