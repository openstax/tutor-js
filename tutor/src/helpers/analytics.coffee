_ = require 'underscore'

Routes = require '../routes'

{CurrentUserStore} = require '../flux/current-user'

DestinationHelper  = require '../helpers/routes-and-destinations'

# generate custom event data for routes
Events =
  viewTaskStep: ({courseId}) ->
    # track all work done on a course
    Analytics.sendEvent 'Course', 'Work', label: courseId

  viewStudentDashboard: ({courseId}) ->
    # compare activity between courses
    Analytics.sendEvent 'Student', 'Dashboard', label: courseId


# Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
Translators =

  dashboard:            ({courseId}) -> "/student/choose-course/#{courseId}"
  viewPractice:         ({courseId}) -> "/student/practice/#{courseId}"
  viewPerformanceForecast:  ({courseId}) -> "/student/performance-forecast/#{courseId}"
  viewStudentDashboard: ({courseId}) -> "/student/dashboard/#{courseId}"

  calendarByDate:        ({courseId}) -> "/teacher/calendar/#{courseId}"
  viewTeacherPerformanceForecast:      ({courseId}) -> "/teacher/performance-forecast/#{courseId}"
  viewScores:            ({courseId}) -> "/teacher/student-scores/#{courseId}"
  courseSettings:        ({courseId}) -> "/teacher/roster/#{courseId}"
  editReading:           ({courseId}) -> "/teacher/assignment/edit/reading/#{courseId}"
  editHomework:          ({courseId}) -> "/teacher/assignment/edit/homework/#{courseId}"
  editExternal:          ({courseId}) -> "/teacher/assignment/edit/external/#{courseId}"
  createReading:         ({courseId}) -> "/teacher/assignment/create/reading/#{courseId}"
  createHomework:        ({courseId}) -> "/teacher/assignment/create/homework/#{courseId}"
  createExternal:        ({courseId}) -> "/teacher/assignment/create/external/#{courseId}"
  calendarViewPlanStats: ({courseId}) -> "/teacher/metrics/quick/#{courseId}"
  reviewTask:            ({courseId}) -> "/teacher/metrics/review/#{courseId}"

  viewReferenceBook:        ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookSection: ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookPage:    ({courseId}) -> "/reference-view/#{courseId}"

  # Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep:         ({courseId}) ->
    role = CurrentUserStore.getCourseRole(courseId, true)
    "/#{role}/task-step/#{courseId}"


GA = undefined

Analytics =

  setTracker: (tracker) -> GA = tracker

  sendPageView: (url) ->
    GA?('send', 'pageview', url)

  onNavigation: (path) ->
    return unless GA

    route = Routes.pathToEntry(path)

    return @sendPageView("/not-found/#{change.path}") unless route

    path = Translators[route.entry.name]?( route.match.params ) or route.match.pathname

    # if we're also going to send custom events then we set the page
    if Events[route.entry.name]
      GA('set', 'page', path)
      Events[route.entry.name]( route.match.params )
      @sendPageView() # url's not needed since it was set before events
    else
      @sendPageView(route.match.pathname)

  sendEvent: (category, action, attrs) ->
    return unless GA
    GA('send', 'event', category, action, attrs.label, attrs.value)

module.exports = Analytics
