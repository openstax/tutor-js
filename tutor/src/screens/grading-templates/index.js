import {
  React, action, PropTypes, observable, observer, computed, styled,
} from 'vendor';
import { Col, Row, Container, Button } from 'react-bootstrap';
import Courses, { Course } from '../../models/courses-map';
import Theme from '../../theme';
import Loading from 'shared/components/loading-animation';
import { GradingTemplates } from '../../models/grading/templates';
import Card from './card';
import DeleteModal from './delete-modal';
import { ScrollToTop } from 'shared';
import CoursePage from '../../components/course-page';
import * as EDIT_TYPES from './editors';
import CourseBreadcrumb from '../../components/course-breadcrumb';

const Heading = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-bottom: 20px;
  border-bottom: 1px solid ${Theme.colors.neutral.pale};
  padding-bottom: 1.6rem;
  > :first-child {
    margin: 0;
    h2 { margin-bottom: 0; }
  }
`;

const Instructions = styled.p`
  font-size: 1.6rem;
  line-height: 2.5rem;
  a {
    text-decoration: underline;
  }
`;

const Templates = styled(CoursePage)`
  min-height: calc(100vh - ${Theme.navbars.top.height});
  background-color: #fff;
`;

export default
@observer
class GradingTemplatesScreen extends React.Component {

  static propTypes = {
    course: PropTypes.instanceOf(Course),
    gradingTemplates: PropTypes.instanceOf(GradingTemplates),
    params: PropTypes.shape({
      courseId: PropTypes.string.isRequired,
    }).isRequired,
  }

  @observable editing = null;
  @observable deleting = null;

  componentDidMount() {
    this.store.fetch();
  }

  @computed get course() {
    return this.props.course || Courses.get(this.props.params.courseId);
  }

  @computed get store() {
    return this.props.gradingTemplates || this.course.gradingTemplates;
  }

  @action.bound onEditComplete() {
    this.editing = null;
  }

  @action.bound onEditTemplate(template) {
    this.editing = template;
  }

  @action.bound onConfirmDelete(template) {
    this.deleting = template;
  }

  @action.bound onDeleteComplete() {
    this.deleting = null;
  }

  @action.bound onDeleteTemplate(template) {
    template.remove();
    this.onDeleteComplete();
  }

  @action.bound onAdd() {
    this.editing = { task_plan_type: 'create' };
  }

  @action.bound onCreateTypeSelection(task_plan_type) {
    this.editing = this.store.newTemplate({ task_plan_type });
  }

  @computed get modal() {
    if (this.editing) {
      const Edit = EDIT_TYPES[this.editing.task_plan_type];
      if (Edit) {
        return (
          <Edit
            template={this.editing}
            onComplete={this.onEditComplete}
            onCreateTypeSelection={this.onCreateTypeSelection}
          />
        );
      }
    }

    if (this.deleting) {
      return (
        <DeleteModal
          onDelete={this.onDeleteTemplate}
          onCancel={this.onDeleteComplete}
          template={this.deleting}
        />
      );
    }

    return null;
  }

  render() {
    return (
      <ScrollToTop>
        <Templates
          course={this.course}
          titleAppearance="light"
        >
          <Container fluid={true}>
            {this.modal}
            <Heading>
              <CourseBreadcrumb
                course={this.course}
                currentTitle="Grading templates"
              />
              {this.course.isWRM && <Button onClick={this.onAdd} size="lg">Add new template</Button>}
            </Heading>
            <Row>
              <Col>
                <Instructions>
                  Manage pre-set submission and grading policy templates here. These
                  templates can be applied to multiple assignments. Template applied
                  to open assignments cannot be edited or deleted. <a href="">Learn more</a>
                </Instructions>
              </Col>
            </Row>
            <Row>
              {this.store.api.isPendingInitialFetch && <Loading message="Fetching templatesâ€¦" />}
              {this.store.undeleted.array.map(template => (
                <Card
                  key={template.id}
                  template={template}
                  onEdit={this.onEditTemplate}
                  onDelete={this.onConfirmDelete}
                />))}
            </Row>
          </Container>
        </Templates>
      </ScrollToTop>
    );
  }

}
