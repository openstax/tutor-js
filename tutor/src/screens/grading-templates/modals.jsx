import { React, PropTypes, styled } from 'vendor';
import { Button, Modal } from 'react-bootstrap';
import { GradingTemplate } from '../../models';

const StyledHeader = styled(Modal.Header)`
  font-weight: bold;

  .close {
    font-size: 3rem;
  }
`;

const StyledBody = styled(Modal.Body)`
  font-size: 1.6rem;
  line-height: 2.5rem;
`;

const ControlsWrapper = styled.div`
  margin-top: 3rem;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`;

const Controls = styled.div`
  .btn + .btn {
    margin-left: 1.5rem;
  }
`;


const DeleteModal = ({ onDelete, onCancel, template }) => {
    return (
        <Modal
            show={true}
            onHide={onCancel}
            backdrop="static"
        >
            <StyledHeader closeButton>
                Delete template?
            </StyledHeader>
            <StyledBody>
                Are you sure you want to permanently
                delete <strong>{template.name}</strong> template?
                You can't undo this action.
                <ControlsWrapper>
                    <Controls>
                        <Button className="delete" variant="default" size="lg" onClick={() => onDelete(template)}>
                            Delete
                        </Button>
                        <Button className="cancel" variant="primary" size="lg" onClick={onCancel}>
                            Cancel
                        </Button>
                    </Controls>
                </ControlsWrapper>
            </StyledBody>
        </Modal>
    );
};
DeleteModal.propTypes = {
    onDelete: PropTypes.func.isRequired,
    onCancel: PropTypes.func.isRequired,
    template: PropTypes.instanceOf(GradingTemplate).isRequired,
};


const NoEditModal = ({ onOk, template }) => {
    return (
        <Modal
            show={true}
            onHide={onOk}
            backdrop="static"
        >
            <StyledHeader className="warning" closeButton>
                Template cannot be edited
            </StyledHeader>
            <StyledBody>
                {template.name} template cannot be edited since
                it is currently in use by assignments
                <ControlsWrapper>
                    <Controls>
                        <Button variant="default" size="lg" onClick={onOk}>
                            Okay
                        </Button>
                    </Controls>
                </ControlsWrapper>
            </StyledBody>
        </Modal>
    );
};
NoEditModal.propTypes = {
    onOk: PropTypes.func.isRequired,
    template: PropTypes.instanceOf(GradingTemplate).isRequired,
};


export { DeleteModal, NoEditModal };
