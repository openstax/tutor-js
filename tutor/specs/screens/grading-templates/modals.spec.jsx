import { React, Factory } from '../../helpers';
import { DeleteModal } from '../../../src/screens/grading-templates/modals';

describe('Delete Modal', function() {
    let props;

    beforeEach(() => {
        props = {
            template: Factory.gradingTemplate({ task_plan_type: 'reading' }),
            onDelete: jest.fn(),
            onCancel: jest.fn(),
        };
    });

    it('matches snapshot', () => {
        expect(<DeleteModal {...props} />).toMatchSnapshot();
    });

    it('deletes and cancels', () => {
        const modal = shallow(<DeleteModal {...props} />);

        modal.find('.cancel').simulate('click');
        expect(props.onCancel).toHaveBeenCalled();

        modal.find('.delete').simulate('click');
        expect(props.onDelete).toHaveBeenCalledWith(props.template);
    });
});
