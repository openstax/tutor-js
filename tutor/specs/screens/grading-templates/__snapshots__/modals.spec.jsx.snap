// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Modal matches snapshot 1`] = `
<DeleteModal
  onCancel={[MockFunction]}
  onDelete={[MockFunction]}
  template={
    GradingTemplate {
      "auto_grading_feedback_on": true,
      "completion_weight": 0.59,
      "correctness_weight": 0.04,
      "default_close_date_offset_days": 1,
      "default_due_date_offset_days": 1,
      "default_due_time": "17:00",
      "default_open_time": "07:00",
      "deleted_at": undefined,
      "has_open_task_plans": undefined,
      "id": 1,
      "late_work_penalty": 0.1,
      "late_work_penalty_applied": "immediately",
      "manual_grading_feedback_on": "publish",
      "map": undefined,
      "name": "Default Reading",
      "task_plan_type": "reading",
    }
  }
/>
`;
